classes/LeadProcessor.cls

```diff
@@ -0,0 +1,36 @@
+global class LeadProcessor implements Database.Batchable<SObject>, Schedulable {
+    
+    global  Integer recordProcesses = 0; 
+    
+    global Database.QueryLocator start(Database.BatchableContext bc){
+        
+        return Database.getQueryLocator('SELECT Id, LeadSource FROM Lead' );
```

Comment:
@Vardhman-Fexle-Trainee, we should add the check for LeadSource != 'Dreamforce' in this query after that we can remove the check from execute method